devtools/x/src/cargo/selected_package.rs

```diff
@@ -65,13 +65,18 @@ impl SelectedPackageArgs {
 
         // Intersect with --changed-since if specified.
         if let Some(base) = &self.changed_since {
-            let affected_set = changed_since_impl(&xctx, &base)?;
-
-            includes = includes.intersection(
-                affected_set
-                    .packages(DependencyDirection::Forward)
-                    .map(|package| package.name()),
-            );
+            if let Ok(git_cli) = xctx.core().git_cli() {
+                let affected_set = changed_since_impl(git_cli, &xctx, &base)?;
+                includes = includes.intersection(
+                    affected_set
+                        .packages(DependencyDirection::Forward)
+                        .map(|package| package.name()),
+                );
+            } else {
+                return Err(anyhow!(
+                    "May only use --changes-since if working in a local git repository."
+                ));
+            }
```

Comment:
Could you replace this with `with_context` so that the chain of errors is maintained? Also it's `--changed-since`.

Reply:
This one didn't end up in typed conflicts.